panic in mb production slowdown code

### Description

```
{"level":"WARN","timestamp":"2025-05-08T21:27:23.770Z","msg":"MiniblockProducer: jobStart: Error creating new miniblock proposal","instanceId":"WyGVpbZvEBzQ","mode":"full","nodeType":"stream","nodeAddress":"0x81aB4c6C8d3f7F4cE829f0B0bcC4D5F8D82816eD","streamId":"a5b4681f09f344d43e20ed7dd289e7c2bf47a579810000000000000000000000","err":{"message":"(8:RESOURCE_EXHAUSTED) mbJob.shouldContinue: candidate production is slowed down","kind":"RESOURCE_EXHAUSTED","candidateCount":425,"blockNum":15300786,"streamId":"a5b4681f09f344d43e20ed7dd289e7c2bf47a579810000000000000000000000","miniblockGeneration":350}}
panic: runtime error: integer divide by zero

goroutine 96838 [running]:
github.com/towns-protocol/towns/core/node/events.skipCandidate(...)
  github.com/towns-protocol/towns/core/node/events/miniblock_job.go:50
github.com/towns-protocol/towns/core/node/events.(*mbJob).shouldContinue(0xc03a4c7380, {0x1eac488, 0xc0006edd10}, 0xe978b2)
  github.com/towns-protocol/towns/core/node/events/miniblock_job.go:68 +0x2e9
github.com/towns-protocol/towns/core/node/events.(*mbJob).produceCandidate(0xc03a4c7380, {0x1eac488, 0xc0006edd10}, 0xc035a757a0?)
  github.com/towns-protocol/towns/core/node/events/miniblock_job.go:77 +0x37
github.com/towns-protocol/towns/core/node/events.(*miniblockProducer).jobStart(0xc0010aa000, {0x1eac488, 0xc0006edd10}, 0xc03a4c7380, 0xe978b2)
  github.com/towns-protocol/towns/core/node/events/miniblock_producer.go:329 +0x79
created by github.com/towns-protocol/towns/core/node/events.(*miniblockProducer).trySchedule in goroutine 96173
  github.com/towns-protocol/towns/core/node/events/miniblock_producer.go:235 +0x154

```

--- a/core/node/events/miniblock_job.go
+++ b/core/node/events/miniblock_job.go
@@ -39,13 +39,24 @@ func skipCandidate(candidateCount int, blockNum crypto.BlockNumber) bool {
 		return false
 	}
 
-	if candidateCount < 10 {
+	cc := candidateCount / 10
+
+	if cc <= 0 {
 		return false
+	}
+
+	if cc > 10 {
+		cc = 10
 	}
 
 	// slow down candidate production by 2x for every 10 candidates for up to 450x slowdown.
 	// i.e. 900 seconds -> once every 15 minutes.
-	slowDownFactor := min(1<<(candidateCount/10), 450)
+	slowDownFactor := min(1<<cc, 450)
+
+	// Really shouldn't happen, but just in case.
+	if slowDownFactor <= 1 {
+		return false
+	}
 
 	return uint64(blockNum)%uint64(slowDownFactor) != 0
 }

